File: models/src/main/java/ar/edu/itba/paw/models/Country.java
package ar.edu.itba.paw.models;

import javax.persistence.*;

@Entity
@Table(
        name = "system_country",
        indexes = {
                @Index(columnList = "country_id", name = "system_country_country_id_uindex", unique = true),
        }
)
public class Country extends GenericModel<String> {
    @Id
    @Column(name = "country_id")
    private String id;
    @Column(name = "name", nullable = false)
    private String name;

    @Override
    public String getId() {
        return this.id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    protected boolean isSameType(Object o) {
        return o instanceof Country;
    }
}

File: webapp/src/main/vueapp/src/logic/models/Country.ts
import {GenericEntity} from '~/logic/models/utils/GenericEntity';

export class Country extends GenericEntity<Country> {
    private _id: string;
    private _name: string;

    public get id(): string {
        return this._id;
    }

    public set id(value: string) {
        this._id = value;
    }

    public get name(): string {
        return this._name;
    }

    public set name(value: string) {
        this._name = value;
    }
}
